source/docs/software/basic-programming/reading-stacktraces.rst

```diff
@@ -0,0 +1,553 @@
+
+      Reading the stack trace, you can see that the issue happened inside of the ``robotInit()`` function, on line 24, and the exception involved "Null Pointer".
+
+      By going to line 24, you can see there is only one thing which could be null - ``armMotorCtrl``. Looking further up, you can see that the ``armMotorCtrl`` object is declared, but never instantiated.
+
+      Alternatively, you can step through lines of code with the single step debugger, and stop when you hit line 24. Inspecting the ``armMotorCtrl`` object at that point would show that it is null.
+
+   .. group-tab:: C++
+
+      .. code-block:: text
+
+         Exception has occurred: W32/0xc0000005
+         Unhandled exception thrown: read access violation.
+         this->motorRef was nullptr.
+
+      In Simulation, this will show up in a debugger window that points to line 20 in the above buggy code.
+
+      You can view the full stack trace by clicking the debugger tab in VS Code:
+
+      .. image:: images/reading-stacktraces/cpp_null_stacktrace.png
+         :alt: Stack Trace associated with a null-related error
+
+      The error is specific - our member variable ``motorRef`` was declared, but never assigned a value. Therefor, when we attempt to use it to call a method using the ``->`` operator, the exception occurs.
+
+      The exception states its type was ``nullptr``.
+
+Fixing Null Object Issues
+"""""""""""""""""""""""""
+
+Generally, you will want to ensure each reference has been initialized before using it. In this case, there is a missing line of code to instantiate the ``armMotorCtrl`` before calling the ``setInverted()`` method.
+
+A functional implementation could look like this:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 19
+
+            PWMSparkMax armMotorCtrl;
+
+            @Override
+            public void robotInit() {
+
+                armMotorCtrl = new PWMSparkMax(0);
+                armMotorCtrl.setInverted(true);
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+         :lineno-start: 17
+
+         class Robot : public frc::TimedRobot {
+            public:
+               void RobotInit() override {
+                  motorRef = &m_armMotor;
+                  motorRef->SetInverted(false);
+               }
+
+            private:
+               frc::PWMVictorSPX m_armMotor{0};
+               frc::PWMVictorSPX * motorRef;
+         };
+
+
+
+Divide by Zero
+^^^^^^^^^^^^^^
+
+It is not generally possible to divide an integer by zero, and expect reasonable results. Most processors (including the roboRIO) will raise an Unhandled Exception.
+
+For example, consider the following code:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 18
+
+            int armLengthRatio;
```

Comment:
This behavior of ints being initialized to zero is not exactly clear so I think we should explicitly initialize it.